primitives/runtime/src/lib.rs

```diff
@@ -902,6 +902,27 @@ impl<R> TransactionOutcome<R> {
 	}
 }
 
+/// Statistics of a block returned by the `dev_getBlockStats` RPC.
+#[derive(Eq, PartialEq, Clone, Copy, Encode, Decode, Debug, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "std", serde(rename_all = "camelCase"))]
+pub struct BlockStats {
+	/// The length in bytes of the storage proof produced by executing the block.
+	pub witness_len: u64,
+	/// The length in bytes of the storage proof after it was compacted.
+	pub witness_compact_len: u64,
+	/// Length of the block in bytes.
+	///
+	/// This information could also be acquired by downloading the whole block. This is merely
+	/// a convenience to keep the traffic between client and node low.
+	pub block_len: u64,
+	/// Number of extrinsics in the block.
+	///
+	/// This information could also be acquired by downloading the whole block. This is merely
+	/// a convenience to keep the traffic between client and node low.
+	pub num_extrinsics: u64,
```

Comment:
For a dev api. Do we really need to keep the traffic low? Sounds like some "over optimization"

Reply:
I would rather argue that leaving it out would be over optimization: It is very easy to include this data. If we'd leave it out the client needs to refetch the block which makes the RPC client more complicated and brittle.

The reason for leaving it out would be to deduplicate. Deduplication is just another form of optimization :)

Reply:
I can update the comment.